src/RejectedPromise.php

```diff
@@ -34,8 +34,8 @@ public function then(callable $onFulfilled = null, callable $onRejected = null)
 
         try {
             return new FulfilledPromise($onRejected($this->exception));
```

Comment:
i think this moves the BC break from our own code to the callable. if the onRejected callback has a type declaration, the fatal error would now happen while calling the callback

Reply:
yes, correct. but thats why i wonder if we should do the widening or sharpen that a rejected promise can be about only a subset of exceptions.